pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -242,16 +285,37 @@ func (e *DockerRegistryServiceController) getDockerRegistryLocations() []string
 		return []string{}
 	}
 	service := obj.(*kapi.Service)
+	result := []string{}
 
 	hasClusterIP := (len(service.Spec.ClusterIP) > 0) && (net.ParseIP(service.Spec.ClusterIP) != nil)
 	if hasClusterIP && len(service.Spec.Ports) > 0 {
-		return []string{
+		result = append(result, []string{
 			net.JoinHostPort(service.Spec.ClusterIP, fmt.Sprintf("%d", service.Spec.Ports[0].Port)),
 			net.JoinHostPort(fmt.Sprintf("%s.%s.svc", service.Name, service.Namespace), fmt.Sprintf("%d", service.Spec.Ports[0].Port)),
+		}...)
+	}
+
+	// Generate dockercfg entry for value of OPENSHIFT_DEFAULT_REGISTRY (if set)
```

Comment:
> You expose the docker-registry via Route (registry.foo.local) -> we generate entry for registry.foo.local (with no ports, assuming docker will figure out whether it is 80 or 443).

Check the docs.  I'm not sure it will.  @pweil- don't I have to explicitly list my ports?